.github/workflows/main.yml

```diff
@@ -0,0 +1,126 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+      - '*_maintenance'
+  pull_request:
+    branches:
+      - master
+      - '*_maintenance'
+
+jobs:
+
+  build:
+
+    strategy:
+
+      # Don't cancel other jobs in the build matrix if one job fails.
+      fail-fast: false
+
+      matrix:
+
+        # Rather than generate all permutations of various settings,
+        # we want to explicitly list each of the variants we want to
+        # test. We can use `name` to declare the names of our variants,
+        # and then use `include` to define their settings.
+
+        name: [
+          linux,
+          linux-debug,
+          macos,
+        ]
+
+        include:
+
+          - name: linux
+            os: ubuntu-16.04
+            buildType: RELEASE
+            containerImage: gafferhq/build:1.1.0
+            dependenciesURL: https://github.com/GafferHQ/dependencies/releases/download/1.3.0/gafferDependencies-1.3.0-linux.tar.gz
+            # GitHub container builds run as root. This causes failures for tests that
+            # assert that filesystem permissions are respected, because root doesn't
+            # respect permissions. So we run the final test suite as a dedicated
+            # test user rather than as root.
+            testRunner: su testUser -c
+            sconsCacheMegabytes: 400
+
+          - name: linux-debug
+            os: ubuntu-16.04
+            buildType: DEBUG
+            containerImage: gafferhq/build:1.1.0
+            dependenciesURL: https://github.com/GafferHQ/dependencies/releases/download/1.3.0/gafferDependencies-1.3.0-linux.tar.gz
+            testRunner: su testUser -c
+            # Debug builds are ludicrously big, so we must use a larger cache
+            # limit. In practice this compresses down to 4-500Mb.
+            sconsCacheMegabytes: 2500
+
+          - name: macos
+            os: macos-10.15
+            buildType: RELEASE
+            containerImage:
+            dependenciesURL: https://github.com/GafferHQ/dependencies/releases/download/1.3.0/gafferDependencies-1.3.0-osx.tar.gz
+            testRunner: bash -c
+            sconsCacheMegabytes: 400
+
+    runs-on: ${{ matrix.os }}
+
+    container: ${{ matrix.containerImage }}
+
+    env:
+      DISPLAY: ":99.0"
+      ARNOLD_LICENSE_ORDER: none # Don't waste time looking for a license that doesn't exist
+
+    steps:
+
+    - uses: actions/checkout@v2
+
+    - name: Install toolchain (MacOS)
+      # Prefer `pip install` where possible because it is faster
+      # than `brew install`.
+      run: |
+        sudo pip install scons
```

Comment:
Added in e3d6145, but I don't think it'll do what you think it will. I _think_ it's `pip` that will decide whether to install for Python 2 or 3, and locking SCons to `3.1.2` will have no effect on that.

Reply:
Cool, thanks. Yeah - thats cool, as hopefully our `pip` will install the right one for itself. Once the base image python changes, then I'm sure we'll be stuffed in other more entertaining ways. At least not using `brew` takes one other meddling party out of the equation. I was just wanting to guard against a future scons release, that broke compatibility, as we've seen with other packages that are now no longer testing against `2.x`.